include/nbl/system/ILogger.h

```diff
@@ -0,0 +1,64 @@
+#include <string>
+#include <cstdint>
+#include <chrono>
+#include <cassert>
+#include <sstream>
+#include <iomanip>
+namespace nbl::system
+{
+	class ILogger
+	{
+	public:
+		enum E_LOG_LEVEL : uint8_t
+		{
+			ELL_NONE = 0,
+			ELL_DEBUG = 1,
+			ELL_INFO = 2,
+			ELL_WARNING = 4,
+			ELL_PERFORMANCE = 8,
+			ELL_ERROR = 16
+		};
+
+		virtual void log(const std::string_view& fmtString, E_LOG_LEVEL logLevel, ...) = 0;
+		virtual void log(const std::wstring_view& fmtString, E_LOG_LEVEL logLevel, ...) = 0;
+	protected:
+		virtual std::string constructLogString(const std::string_view& fmtString, E_LOG_LEVEL logLevel, va_list l)
+		{
+			using namespace std::literals;
+			auto currentTime = std::chrono::system_clock::now();
+			const std::time_t t = std::chrono::system_clock::to_time_t(currentTime);
+			std::stringstream ss;
+			ss << '[' << std::put_time(std::localtime(&t), "%F %T.\n") << ']';
```

Comment:
how accurate is this time?

Reply:
Not very accurate, and it looks like you would want to write you own timer for each platform, but i assume it will be possible to use Nabla timer when it's done?

Reply:
nah I mean miliseconds or microseconds?

Reply:
This one wil give you the date in this format: `YYYY-MM-DD HH-MM-SS`, but i can make it with microseconds

Reply:
plz do icroseconds can just be seconds with 6 decimal places attached